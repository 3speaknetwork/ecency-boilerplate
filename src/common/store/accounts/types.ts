import { ExtendedAccount } from "@esteemapp/dhive";

export interface Account extends ExtendedAccount {}

export interface State {
  data: Account;
  loading: boolean;
  error: boolean;
}

export enum ActionTypes {
  FETCH = "@account/FETCH",
  FETCHED = "@account/FETCHED",
  FETCH_ERROR = "@account/FETCH_ERROR"
}

export interface FetchAction {
  type: ActionTypes.FETCH;
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  data: Account;
}

export interface FetchErrorAction {
  type: ActionTypes.FETCH_ERROR;
}

export type Actions = FetchAction | FetchedAction | FetchErrorAction;