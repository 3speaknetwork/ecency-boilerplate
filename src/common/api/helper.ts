import isElectron from "../util/is-electron";

import defaults from "../constants/defaults.json";

export const apiBase = (endpoint: string): string => {
    let base = '';

    if (isElectron()) {
        if (process.env.NODE_ENV === "development") {
            base = "http://localhost:3000";
        } else {
            base = defaults.base;
        }
    }

    return `${base}${endpoint}`;
}
