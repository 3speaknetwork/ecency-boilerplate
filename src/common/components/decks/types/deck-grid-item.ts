interface BaseDeckGridItem {
  key: number;
  type: "u" | "ac" | "co" | "w" | "n" | "tr" | "to" | "s" | "cu";
  settings: unknown;
}

export interface UserDeckGridItem extends BaseDeckGridItem {
  type: "u";
  settings: {
    username: string;
    contentType: string;
  };
}

export type DeckGridItem = BaseDeckGridItem | UserDeckGridItem;
