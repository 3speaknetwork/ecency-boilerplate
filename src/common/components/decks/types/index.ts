export * from "./deck-grid-item";
export * from "./deck-grid";
