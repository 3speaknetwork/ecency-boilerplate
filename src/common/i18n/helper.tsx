import React from "react";

import {_t} from "./index";

interface Props {
    k: string,
    args: {},
    children: JSX.Element
}

export class Tsx extends React.Component<Props> {
    render() {
        const {k, args} = this.props;
        const {children} = this.props;
        return React.cloneElement(children, {dangerouslySetInnerHTML: {__html: _t(k, args)}});
    }
}
