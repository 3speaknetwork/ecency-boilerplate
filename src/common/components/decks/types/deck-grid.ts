import { DeckGridItem } from "./deck-grid-item";

export interface DeckGrid {
  columns: DeckGridItem[];
}
