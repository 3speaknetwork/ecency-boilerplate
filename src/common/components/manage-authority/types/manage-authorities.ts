import { Account } from "../../../store/accounts/types";

export const Keytype = {
  Owner: "owner",
  Active: "active",
  Posting: "posting"
};

export interface PublicKeys {
  publicOwnerKey?: any;
  publicActiveKey?: any;
  publicPostingKey?: any;
}

export interface AccountDataType {
  postingsAuthority: Array<any>;
  posting: Array<any>;
  owner: Array<any>;
  active: Array<any>;
  weight: number;
  memokey: string;
  account: Account;
  PublicKeys: {
    publicOwnerKey: any;
    publicActiveKey: any;
    publicPostingKey: any;
  };
}

export const actionType = {
  Revoke: "revoke"
};
