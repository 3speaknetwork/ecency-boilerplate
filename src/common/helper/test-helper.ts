import { State as GlobalState, EntryFilter, Theme, ListStyle } from "../store/global/types";
import { Entry } from "../store/entries/types";
import { Community } from "../store/community/types";
import { DynamicProps } from "../store/dynamic-props/types";
import { Vote, VestingDelegation } from "../../common/api/hive";

export const globalInstance: GlobalState = {
  filter: EntryFilter.hot,
  tag: "",
  theme: Theme.day,
  listStyle: ListStyle.row,
  intro: true,
  currency: "usd",
  currencyRate: 1,
  currencySymbol: "$",
};

export const entryInstance1: Entry = {
  post_id: 86342505,
  author: "good-karma",
  permlink: "awesome-hive",
  category: "hive",
  title: "Awesome Hive",
  body:
    "Hey developers,\n\n![awesome-hive.png](https://images.esteem.app/DQmetNmv6rtcRxueXJSNm7ErLLKNYGJETxsNSoehn6xk9BC/awesome-hive.png)\n\nI just created [Awesome-Hive](https://github.com/ledgerconnect/awesome-hive), list of services and apps on Hive.\n\nIf you don't know about Awesome project, take a look at here: https://project-awesome.org/\n\nProject contains list of services and links to various tools and libraries. Quite useful if you are developing or learning some new programming languages or skills. So we now have `awesome-hive` list, feel free to submit your pull request to add your apps, services. \n\nI will submit PR to project awesome in 30 days to list Hive in decentralized systems section. One of the requirements, awesome list should be around for more than 30 days. So I ask anyone with service, tools, dapps to submit pull request to https://github.com/ledgerconnect/awesome-hive so we have complete list of services by end of June.\n\nIn 30 days, we will PR to official project awesome list. More awareness about awesome Hive! Meanwhile, let's complete `awesome-hive` together! \ud83d\ude0e \ud83d\ude47 \ud83d\ude4f \n\n## Hive on!",
  json_metadata: {
    tags: ["hive", "hiveio", "awesome", "devs", "development", "list"],
    app: "esteem/2.2.7-surfer",
  },
  created: "2020-06-01T05:53:33",
  updated: "2020-06-01T14:41:15",
  depth: 0,
  children: 31,
  net_rshares: 234863836438976,
  is_paidout: false,
  payout_at: "2020-06-08T05:53:33",
  payout: 125.095,
  pending_payout_value: "125.095 HBD",
  author_payout_value: "0.000 HBD",
  curator_payout_value: "0.000 HBD",
  promoted: "0.000 HBD",
  replies: [],
  active_votes: [
    {
      voter: "user1",
      rshares: "34273117581576",
    },
    {
      voter: "user2",
      rshares: "2269348064337",
    },
    {
      voter: "user3",
      rshares: "19969726098",
    },
    {
      voter: "user4",
      rshares: "725359024516",
    },
    {
      voter: "user5",
      rshares: "24029616493",
    },
    {
      voter: "user6",
      rshares: "1485954337",
    },
    {
      voter: "user7",
      rshares: "1281560607198",
    },
    {
      voter: "user8",
      rshares: "554323413016",
    },
    {
      voter: "user9",
      rshares: "140063022903",
    },
    {
      voter: "user10",
      rshares: "542424211",
    },
  ],
  author_reputation: 76.58,
  stats: {
    hide: false,
    gray: false,
    total_votes: 10,
    flag_weight: 1.0,
  },
  beneficiaries: [
    {
      account: "esteemapp",
      weight: 300,
    },
  ],
  max_accepted_payout: "1000000.000 HBD",
  percent_steem_dollars: 10000,
  url: "/hive/@good-karma/awesome-hive",
  blacklists: [],
};

export const votesInstance1: Vote[] = [
  {
    voter: "user1",
    weight: 40289,
    rshares: "84880616925",
    percent: 40,
    time: "2020-06-14T15:19:51",
  },
  {
    voter: "user2",
    weight: 0,
    rshares: "39117096",
    percent: 10000,
    time: "2020-06-14T15:15:15",
  },
];

export const vestingDelegationInstance1: VestingDelegation[] = [
  {
    id: 1327252,
    delegator: "esteemapp",
    delegatee: "user1",
    vesting_shares: "14848.000000 VESTS",
    min_delegation_time: "2020-02-24T09:53:57",
  },
  {
    id: 1329459,
    delegator: "esteemapp",
    delegatee: "user2",
    vesting_shares: "191778.000000 VESTS",
    min_delegation_time: "2020-03-13T08:34:57",
  },
  {
    id: 1325910,
    delegator: "esteemapp",
    delegatee: "user3",
    vesting_shares: "29326.000000 VESTS",
    min_delegation_time: "2020-02-18T18:32:06",
  },
  {
    id: 1325312,
    delegator: "esteemapp",
    delegatee: "user4",
    vesting_shares: "27751.000000 VESTS",
    min_delegation_time: "2020-02-17T09:28:12",
  },
  {
    id: 1330486,
    delegator: "joythewanderer",
    delegatee: "user5",
    vesting_shares: "1957.888055 VESTS",
    min_delegation_time: "2020-03-24T12:42:39",
  },
];

export const discussionInstace1: Entry[] = [
  {
    post_id: 86423250,
    author: "esteemapp",
    permlink: "esteem-discord-monthly-giveaway-winners-21",
    category: "esteem",
    title: "Esteem Discord Monthly Giveaway Winners #21",
    body: "...",
    json_metadata: {},
    created: "2020-06-06T05:37:51",
    updated: "2020-06-06T05:37:51",
    depth: 0,
    children: 9,
    net_rshares: 23953037130835,
    is_paidout: false,
    payout_at: "2020-06-13T05:37:51",
    payout: 11.432,
    pending_payout_value: "11.432 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [],
    author_reputation: 75.6,
    stats: { hide: false, gray: false, total_votes: 180, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21",
    blacklists: [],
  },
  {
    post_id: 86437193,
    author: "forykw",
    permlink: "re-esteemapp-202067t12246786z",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "Prizes have consistently been very effective from @esteemapp nicely done!",
    json_metadata: {},
    created: "2020-06-07T00:02:48",
    updated: "2020-06-07T00:02:48",
    depth: 1,
    children: 0,
    net_rshares: 0,
    is_paidout: false,
    payout_at: "2020-06-14T00:02:48",
    payout: 0,
    pending_payout_value: "0.000 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [],
    author_reputation: 64.87,
    stats: { hide: false, gray: false, total_votes: 0, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@forykw/re-esteemapp-202067t12246786z",
    blacklists: [],
  },
  {
    post_id: 86427694,
    author: "brittandjosie",
    permlink: "re-esteemapp-qbi6t3",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "@ardpien congrats on being one of the winners ",
    json_metadata: { tags: ["esteem"], app: "peakd/2020.05.5" },
    created: "2020-06-06T12:03:03",
    updated: "2020-06-06T12:03:03",
    depth: 1,
    children: 1,
    net_rshares: 0,
    is_paidout: false,
    payout_at: "2020-06-13T12:03:03",
    payout: 0,
    pending_payout_value: "0.000 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: ["ardpien/re-brittandjosie-qbjoep"],
    active_votes: [],
    author_reputation: 71.78,
    stats: { hide: false, gray: false, total_votes: 0, flag_weight: 0 },
    beneficiaries: [],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@brittandjosie/re-esteemapp-qbi6t3",
    blacklists: [],
  },
  {
    post_id: 86426904,
    author: "ardpien",
    permlink: "qbi303",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body:
      "ðŸ˜ƒ I am mentioned here. That's great! All thanks to ESTEEM for creating an awesome Discord server for us to talk in there and participate in activities. Thank You ESTEEM team :) and @good-karma.",
    json_metadata: {},
    created: "2020-06-06T10:40:54",
    updated: "2020-06-06T10:40:54",
    depth: 1,
    children: 0,
    net_rshares: 114901996313,
    is_paidout: false,
    payout_at: "2020-06-13T10:40:54",
    payout: 0.031,
    pending_payout_value: "0.031 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [{ voter: "jznsamuel", rshares: "114901996313" }],
    author_reputation: 47.13,
    stats: { hide: false, gray: false, total_votes: 1, flag_weight: 0 },
    beneficiaries: [],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@ardpien/qbi303",
    blacklists: [],
  },
  {
    post_id: 86426429,
    author: "iliyan90",
    permlink: "re-esteemapp-202066t124616772z",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "@esteemapp\nBug report [click here](https://esteem.app/esteem/@iliyan90/bug-report-to-esteemapp)",
    json_metadata: {},
    created: "2020-06-06T09:46:15",
    updated: "2020-06-06T09:46:36",
    depth: 1,
    children: 0,
    net_rshares: 0,
    is_paidout: false,
    payout_at: "2020-06-13T09:46:15",
    payout: 0,
    pending_payout_value: "0.000 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [],
    author_reputation: 60.22,
    stats: { hide: false, gray: false, total_votes: 0, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@iliyan90/re-esteemapp-202066t124616772z",
    blacklists: [],
  },
  {
    post_id: 86425273,
    author: "trincowski",
    permlink: "re-esteemapp-202066t84137915z",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "Awesome. Thank you very much.",
    json_metadata: {},
    created: "2020-06-06T07:41:39",
    updated: "2020-06-06T07:41:39",
    depth: 1,
    children: 0,
    net_rshares: 116060350556,
    is_paidout: false,
    payout_at: "2020-06-13T07:41:39",
    payout: 0.032,
    pending_payout_value: "0.032 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [{ voter: "jznsamuel", rshares: "116060350556" }],
    author_reputation: 68.52,
    stats: { hide: false, gray: false, total_votes: 1, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@trincowski/re-esteemapp-202066t84137915z",
    blacklists: [],
  },
  {
    post_id: 86423472,
    author: "foxkoit",
    permlink: "re-esteemapp-202066t8590757z",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "Thank you ðŸ˜€ðŸ˜‡ðŸ˜‡",
    json_metadata: {},
    created: "2020-06-06T05:59:03",
    updated: "2020-06-06T05:59:03",
    depth: 1,
    children: 0,
    net_rshares: 117230405346,
    is_paidout: false,
    payout_at: "2020-06-13T05:59:03",
    payout: 0.032,
    pending_payout_value: "0.032 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [{ voter: "jznsamuel", rshares: "117230405346" }],
    author_reputation: 71.51,
    stats: { hide: false, gray: false, total_votes: 1, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@foxkoit/re-esteemapp-202066t8590757z",
    blacklists: [],
  },
  {
    post_id: 86423446,
    author: "irisworld",
    permlink: "re-esteemapp-202066t14588243z",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "yahoo! thanks a lot!",
    json_metadata: {},
    created: "2020-06-06T05:58:09",
    updated: "2020-06-06T05:58:09",
    depth: 1,
    children: 0,
    net_rshares: 265974487101,
    is_paidout: false,
    payout_at: "2020-06-13T05:58:09",
    payout: 0.075,
    pending_payout_value: "0.075 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [],
    author_reputation: 65.01,
    stats: { hide: false, gray: false, total_votes: 2, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@irisworld/re-esteemapp-202066t14588243z",
    blacklists: [],
  },
  {
    post_id: 86423439,
    author: "behiver",
    permlink: "re-esteemapp-202066t85739415z",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body:
      "Great to see my name on the Giveaway and I like quite much engaging on Discord and find out what others are doing in the HIVE space. A source of news, common hobbies and other topics that rise up your day.",
    json_metadata: {},
    created: "2020-06-06T05:57:39",
    updated: "2020-06-06T05:57:39",
    depth: 1,
    children: 0,
    net_rshares: 273072542004,
    is_paidout: false,
    payout_at: "2020-06-13T05:57:39",
    payout: 0.077,
    pending_payout_value: "0.077 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [
      { voter: "foxkoit", rshares: "154659337454" },
      { voter: "jznsamuel", rshares: "118413204550" },
    ],
    author_reputation: 57.66,
    stats: { hide: false, gray: false, total_votes: 2, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@behiver/re-esteemapp-202066t85739415z",
    blacklists: [],
  },
  {
    post_id: 86440916,
    author: "ardpien",
    permlink: "re-brittandjosie-qbjoep",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "Thank you, Miss. @birttandjosie.",
    json_metadata: { tags: ["esteem"], app: "peakd/2020.05.5" },
    created: "2020-06-07T07:20:51",
    updated: "2020-06-07T07:20:51",
    depth: 2,
    children: 0,
    net_rshares: 0,
    is_paidout: false,
    payout_at: "2020-06-14T07:20:51",
    payout: 0,
    pending_payout_value: "0.000 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [],
    author_reputation: 47.13,
    stats: { hide: false, gray: false, total_votes: 0, flag_weight: 0 },
    beneficiaries: [],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "brittandjosie",
    parent_permlink: "re-esteemapp-qbi6t3",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@ardpien/re-brittandjosie-qbjoep",
    blacklists: [],
  },
];

export const communityInstance1: Community = {
  id: 1369030,
  name: "hive-148441",
  title: "GEMS",
  about: "To highlight true Gems of Hive community with User Retention as primary objective.",
  lang: "en",
  type_id: 1,
  is_nsfw: false,
  subscribers: 4086,
  sum_pending: 17370,
  num_pending: 12481,
  num_authors: 1796,
  created_at: "2020-02-26 11:33:33",
  avatar_url: "",
  description:
    "This is a collective curation community formed by @appreciator, @rocky1 and @upmewhale. You can post your genuine and creative content in this community to get more support.\n\n\nYou can post in languages other than English.\n\nThere are not many restrictions but just few simple ones and we expect you to adhere to them\n\nJoin our discord group to keep yourself updated:\nhttps://discord.gg/n98Kpmm",
  flag_text: "Plagiarism will be highly discouraged.\nHate Speech is not allowed",
  settings: {},
  team: [
    ["hive-148441", "owner", ""],
    ["bluemist", "admin", ""],
  ],
};

export const dynamicPropsIntance1: DynamicProps = {
  hivePerMVests: 513.7574961580294,
  base: 0.252,
  quote: 1,
  fundRewardBalance: 848692.443,
};
