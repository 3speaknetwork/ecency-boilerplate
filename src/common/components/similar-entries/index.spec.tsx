import React from 'react';
import renderer from "react-test-renderer";

import {createBrowserHistory, createLocation} from "history";

import {SimilarEntries} from './index';

import {globalInstance, entryInstance1} from "../../helper/test-helper";
import {Entry} from "../../store/entries/types";

const allOver = () => new Promise((resolve) => setImmediate(resolve));

let TEST_MODE = 0

jest.mock("../../api/bridge", () => ({
    getAccountPosts: () =>
        new Promise((resolve) => {
            const p1: Entry = {...entryInstance1, permlink: "foo1"}
            const p2: Entry = {...entryInstance1, permlink: "foo2"}
            const p3: Entry = {...entryInstance1, permlink: "foo3"}
            const p4: Entry = {...entryInstance1, permlink: "foo4"}
            const p5: Entry = {...entryInstance1, permlink: "foo5"}

            if (TEST_MODE === 0) {
                resolve([]);
            }

            if (TEST_MODE === 1) {
                resolve([p1, p2]);
            }

            if (TEST_MODE === 2) {
                resolve([p1, p2, p3, p4, p5]);
            }
        }),
}));

jest.mock("moment", () => () => ({
    fromNow: () => "3 days ago"
}));


it('(1) No data.', async () => {
    const props = {
        history: createBrowserHistory(),
        location: createLocation({}),
        global: globalInstance,
        entry: entryInstance1
    };

    const component = renderer.create(<SimilarEntries {...props}/>);
    await allOver();
    expect(component.toJSON()).toMatchSnapshot();
});


it('(2) Less than 3 entries. Should render null', async () => {
    TEST_MODE = 1;

    const props = {
        history: createBrowserHistory(),
        location: createLocation({}),
        global: globalInstance,
        entry: entryInstance1
    };

    const component = renderer.create(<SimilarEntries {...props}/>);
    await allOver();
    expect(component.toJSON()).toMatchSnapshot();
});


it('(3) Should render entries', async () => {
    TEST_MODE = 2;

    const props = {
        history: createBrowserHistory(),
        location: createLocation({}),
        global: globalInstance,
        entry: entryInstance1
    };

    const component = renderer.create(<SimilarEntries {...props}/>);
    await allOver();
    expect(component.toJSON()).toMatchSnapshot();
});
