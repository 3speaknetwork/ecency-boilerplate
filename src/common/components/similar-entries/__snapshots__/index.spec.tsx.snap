// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) No data. 1`] = `null`;

exports[`(2) Less than 3 entries. Should render null 1`] = `null`;

exports[`(3) Should render entries 1`] = `
<div
  className="similar-entries-list"
>
  <div
    className="similar-entries-list-header"
  >
    <div
      className="list-header-text"
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M7,5H21V7H7V5M7,13V11H21V13H7M4,4.5A1.5,1.5 0 0,1 5.5,6A1.5,1.5 0 0,1 4,7.5A1.5,1.5 0 0,1 2.5,6A1.5,1.5 0 0,1 4,4.5M4,10.5A1.5,1.5 0 0,1 5.5,12A1.5,1.5 0 0,1 4,13.5A1.5,1.5 0 0,1 2.5,12A1.5,1.5 0 0,1 4,10.5M7,19V17H21V19H7M4,16.5A1.5,1.5 0 0,1 5.5,18A1.5,1.5 0 0,1 4,19.5A1.5,1.5 0 0,1 2.5,18A1.5,1.5 0 0,1 4,16.5Z"
          fill="currentColor"
        />
      </svg>
       
      More from @good-karma
    </div>
  </div>
  <div
    className="similar-entries-list-body"
  >
    <div
      className="similar-entries-list-item"
    >
      <a
        href="/foo/@good-karma/foo1"
        onClick={[Function]}
      >
        <div
          className="item-title"
        >
          Awesome Hive
        </div>
        <div
          className="item-image"
        >
          <img
            alt="Awesome Hive"
            onError={[Function]}
            src="https://images.ecency.com/p/HNWT6DgoBc1692QWn5trsGLxi44TrGuNZJMEdibBBbUKuho3EaW43JV4QzaQ258PEGNV2u43KAhU8r3Vjno4cBP8JeTFkbS2SCMjZZPprKzfMSdteRxSZxuPku4?format=match&mode=fit&width=600&height=500"
          />
        </div>
        <div
          className="item-footer"
        >
          <span
            className="item-footer-date"
          >
            3 days ago
          </span>
          <span
            className="item-footer-comment-count"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M18,8H6V6H18V8M18,11H6V9H18V11M18,14H6V12H18V14M22,4A2,2 0 0,0 20,2H4A2,2 0 0,0 2,4V16A2,2 0 0,0 4,18H18L22,22V4Z"
                fill="currentColor"
              />
            </svg>
             
            31
          </span>
        </div>
      </a>
    </div>
    <div
      className="similar-entries-list-item"
    >
      <a
        href="/foo/@good-karma/foo2"
        onClick={[Function]}
      >
        <div
          className="item-title"
        >
          Awesome Hive
        </div>
        <div
          className="item-image"
        >
          <img
            alt="Awesome Hive"
            onError={[Function]}
            src="https://images.ecency.com/p/HNWT6DgoBc1692QWn5trsGLxi44TrGuNZJMEdibBBbUKuho3EaW43JV4QzaQ258PEGNV2u43KAhU8r3Vjno4cBP8JeTFkbS2SCMjZZPprKzfMSdteRxSZxuPku4?format=match&mode=fit&width=600&height=500"
          />
        </div>
        <div
          className="item-footer"
        >
          <span
            className="item-footer-date"
          >
            3 days ago
          </span>
          <span
            className="item-footer-comment-count"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M18,8H6V6H18V8M18,11H6V9H18V11M18,14H6V12H18V14M22,4A2,2 0 0,0 20,2H4A2,2 0 0,0 2,4V16A2,2 0 0,0 4,18H18L22,22V4Z"
                fill="currentColor"
              />
            </svg>
             
            31
          </span>
        </div>
      </a>
    </div>
    <div
      className="similar-entries-list-item"
    >
      <a
        href="/foo/@good-karma/foo3"
        onClick={[Function]}
      >
        <div
          className="item-title"
        >
          Awesome Hive
        </div>
        <div
          className="item-image"
        >
          <img
            alt="Awesome Hive"
            onError={[Function]}
            src="https://images.ecency.com/p/HNWT6DgoBc1692QWn5trsGLxi44TrGuNZJMEdibBBbUKuho3EaW43JV4QzaQ258PEGNV2u43KAhU8r3Vjno4cBP8JeTFkbS2SCMjZZPprKzfMSdteRxSZxuPku4?format=match&mode=fit&width=600&height=500"
          />
        </div>
        <div
          className="item-footer"
        >
          <span
            className="item-footer-date"
          >
            3 days ago
          </span>
          <span
            className="item-footer-comment-count"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M18,8H6V6H18V8M18,11H6V9H18V11M18,14H6V12H18V14M22,4A2,2 0 0,0 20,2H4A2,2 0 0,0 2,4V16A2,2 0 0,0 4,18H18L22,22V4Z"
                fill="currentColor"
              />
            </svg>
             
            31
          </span>
        </div>
      </a>
    </div>
  </div>
</div>
`;
